Source/Scene/modelMaterialsCommon.js

```diff
@@ -0,0 +1,792 @@
+/*global define*/
+define([
+    '../Core/defaultValue',
+    '../Core/defined',
+    '../Renderer/WebGLConstants'
+], function(
+    defaultValue,
+    defined,
+    WebGLConstants) {
+    "use strict";
+
+    function webGLConstantToGlslType(webGLValue) {
+        switch(webGLValue) {
+            case WebGLConstants.FLOAT:
+                return 'float';
+            case WebGLConstants.FLOAT_VEC2:
+                return 'vec2';
+            case WebGLConstants.FLOAT_VEC3:
+                return 'vec3';
+            case WebGLConstants.FLOAT_VEC4:
+                return 'vec4';
+            case WebGLConstants.FLOAT_MAT2:
+                return 'mat2';
+            case WebGLConstants.FLOAT_MAT3:
+                return 'mat3';
+            case WebGLConstants.FLOAT_MAT4:
+                return 'mat4';
+            case WebGLConstants.SAMPLER_2D:
+                return 'sampler2D';
+        }
+    }
+
+    function generateLightParameters(gltf) {
+        var result = {};
+
+        var lights;
+        if (defined(gltf.extensions) && defined(gltf.extensions.KHR_materials_common)) {
+            lights = gltf.extensions.KHR_materials_common.lights;
+        }
+
+        if (defined(lights)) {
+            // Figure out which node references the light
+            var nodes = gltf.nodes;
+            for (var nodeName in nodes) {
+                if (nodes.hasOwnProperty(nodeName)) {
+                    var node = nodes[nodeName];
+                    if (defined(node.extensions) && defined(node.extensions.KHR_materials_common)) {
+                        var nodeLightId = node.extensions.KHR_materials_common.light;
+                        if (defined(nodeLightId) && defined(lights[nodeLightId])) {
+                            lights[nodeLightId].node = nodeName;
```

Comment:
Couldn't more than one node reference the same light, e.g., two directional lights that are headlights on a car.

We should be explicit about this in the extension spec.

CC @tparisi